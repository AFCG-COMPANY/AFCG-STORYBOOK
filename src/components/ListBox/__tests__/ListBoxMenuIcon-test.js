import React from 'react';
import { mount } from 'enzyme';
import ListBox from '../';

describe('ListBoxMenuIcon', () => {
  it('should render', () => {
    const openWrapper = mount(<ListBox.MenuIcon isOpen={true} />);
    const closedWrapper = mount(<ListBox.MenuIcon isOpen={false} />);
    expect(openWrapper).toMatchSnapshot();
    expect(closedWrapper).toMatchSnapshot();
  });

  it('should call `translateWithId` to determine the description', () => {
    const translateWithId = jest.fn(() => 'message');
    mount(<ListBox.MenuIcon isOpen={true} translateWithId={translateWithId} />);
    expect(translateWithId).toHaveBeenCalledWith('close.menu');

    translateWithId.mockClear();

    mount(
      <ListBox.MenuIcon isOpen={false} translateWithId={translateWithId} />
    );
    expect(translateWithId).toHaveBeenCalledWith('open.menu');
  });
});
