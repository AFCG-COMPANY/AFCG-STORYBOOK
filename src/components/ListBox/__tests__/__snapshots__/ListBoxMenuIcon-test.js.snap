// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBoxMenuIcon should render 1`] = `
<ListBoxMenuIcon
  isOpen={true}
  translateWithId={[Function]}
>
  <div
    className="bx--list-box__menu-icon bx--list-box__menu-icon--open"
  >
    <Icon
      alt="Close menu"
      description="Close menu"
      fillRule="evenodd"
      name="caret--down"
      role="img"
    >
      <svg
        alt="Close menu"
        aria-label="Close menu"
        fillRule="evenodd"
        height="5"
        name="caret--down"
        role="img"
        viewBox="0 0 10 5"
        width="10"
      >
        <title>
          Close menu
        </title>
        <path
          d="M0 0l5 4.998L10 0z"
          key="key0"
        />
      </svg>
    </Icon>
  </div>
</ListBoxMenuIcon>
`;

exports[`ListBoxMenuIcon should render 2`] = `
<ListBoxMenuIcon
  isOpen={false}
  translateWithId={[Function]}
>
  <div
    className="bx--list-box__menu-icon"
  >
    <Icon
      alt="Open menu"
      description="Open menu"
      fillRule="evenodd"
      name="caret--down"
      role="img"
    >
      <svg
        alt="Open menu"
        aria-label="Open menu"
        fillRule="evenodd"
        height="5"
        name="caret--down"
        role="img"
        viewBox="0 0 10 5"
        width="10"
      >
        <title>
          Open menu
        </title>
        <path
          d="M0 0l5 4.998L10 0z"
          key="key0"
        />
      </svg>
    </Icon>
  </div>
</ListBoxMenuIcon>
`;
